File: VB/PreviewClickSample/MainWindowViewModel.vb
Imports PreviewClickSample
Imports DevExpress.Xpf.Printing
Imports System.Collections.Generic
Imports System.Windows.Input

Namespace PreviewClickSample
    Public Class MainWindowViewModel
        ReadOnly report As DrillDownReport = New DrillDownReport
        Public Property PreviewModel As XtraReportPreviewModel

        Public Sub New()
            PreviewModel = New XtraReportPreviewModel(report)
            AddHandler PreviewModel.PreviewMouseMove, AddressOf model_PreviewMouseMove
            AddHandler PreviewModel.PreviewClick, AddressOf model_PreviewClick
            report.CreateDocument()
        End Sub

        Private Sub model_PreviewClick(ByVal sender As Object, ByVal e As PreviewClickEventArgs)
            If String.IsNullOrEmpty(e.ElementTag) Then
                Return
            End If
            Dim expandedValues As List(Of Integer) = CType(report.Parameters("expandedValuesParameter").Value, List(Of Integer))
            Dim index As Integer
            If (Not Integer.TryParse(e.ElementTag, index)) Then
                Return
            End If

            If expandedValues.Contains(index) Then
                expandedValues.Remove(index)
            Else
                expandedValues.Add(index)
            End If
            report.CreateDocument(False)
        End Sub

        Private Sub model_PreviewMouseMove(ByVal sender As Object, ByVal e As PreviewClickEventArgs)
            If Not String.IsNullOrEmpty(e.ElementTag) Then
                e.Element.Cursor = Cursors.Hand
            End If
        End Sub
    End Class
End Namespace

File: CS/PreviewClickSample/MainWindowViewModel.cs
using System.Collections.Generic;
using System.Windows.Input;
using DevExpress.Xpf.Printing;
// ...

namespace PreviewClickSample {
    public class MainWindowViewModel {
        readonly DrillDownReport report = new DrillDownReport();

        public XtraReportPreviewModel PreviewModel { get; private set; }

        public MainWindowViewModel() {
            PreviewModel = new XtraReportPreviewModel(report);
            PreviewModel.PreviewMouseMove += model_PreviewMouseMove;
            PreviewModel.PreviewClick += model_PreviewClick;
            report.CreateDocument();
        }

        void model_PreviewClick(object sender, PreviewClickEventArgs e) {
            int index;
            if(string.IsNullOrEmpty(e.ElementTag) || !int.TryParse(e.ElementTag, out index)) {
                return;
            }
            var expandedValues = (List<int>)report.Parameters["expandedValuesParameter"].Value;
            if(expandedValues.Contains(index)) {
                expandedValues.Remove(index);
            } else {
                expandedValues.Add(index);
            }
            report.CreateDocument(false);
        }

        void model_PreviewMouseMove(object sender, PreviewClickEventArgs e) {
            if(!string.IsNullOrEmpty(e.ElementTag)) {
                e.Element.Cursor = Cursors.Hand;
            }
        }
    }
}
